src/react.tsx

```diff
@@ -15,23 +38,36 @@ export function connect<StoreState extends object>(store: StoreDefinition<StoreS
   ): React.ComponentClass<Diff<PropsWithStore, { store: Store<StoreState> }>> {
 
     type State = {
-      store: StoreSnapshot<StoreState>
-      subscription: Subscription
+      store: StoreSnapshotWrapper<StoreState>
     }
 
     return class extends React.Component<Diff<PropsWithStore, { store: Store<StoreState> }>, State> {
       static displayName = `withStore(${getDisplayName(Component)})`
+      const subscriptions: Map<keyof StoreState, Subscription> = new Map()
+
+      const _onGet = (field: keyof StoreState) => {
+        if (this.subscriptions.has(field)) {
+          return
+        }
+        this.subscriptions.set(
+          field,
+          store.on(field).subscribe(
+            (value) => {
+              if (equals(value, this.state.store.get(field))) {
```

Comment:
Now that I'm thinking about it, it might be best to get rid of this `equals` check completely. It was intended as a sanity check to improve performance, but after your change, it's an optimization that's probably best left to consumers. You or I can do it in a follow-up PR so that we don't do too much in this PR.

Reply:
Lets leave it for now